### PR TITLE
Improve instructions on how to run tests for AI agents

### PR DESCRIPTION
**Pull request by y9v**
**What does this PR do?**
This PR aims to improve the identification of a correct test task for AI agents.

**Motivation:**
AI agents (Claude Code, Cursor, etc.) currently try running isolated spec files with `bundle exec rspec ...`, which fails for contrib/integration tests that require specific appraisal Gemfiles. This change teaches the agents the `lib/` → `spec/` → `rake test:TASK_KEY` mapping, including the product-level pattern (namespaced tasks for appsec/profiling/di/ai_guard, flat tasks for tracing contribs), and how to self-serve edge cases by checking the `Rakefile` and `Matrixfile`.

**Change log entry**
None.

**Additional Notes:**
I used AI to write these changes, used AI to test if these instructions actually improve anything.

**How to test the change?**
Try to do some simple change using your favorite coding agent, see if it can correctly identify the test tasks it needs to run.

### CODE REVIEW

File: .cursor/rules/testing.mdc [hunk 1]
```diff
@@ -31,9 +31,39 @@ alwaysApply: false
...
+- **Other**: `test:error_tracking`, `test:opentelemetry`, `test:open_feature`, `test:autoinstrument`, `test:custom_cop`
+
+When any AppSec integration changes, also run: `docker compose run --rm tracer-3.3 bundle exec rake test:appsec:integration`
+
+See `CLAUDE.md` "Running Tests" for more detail. When unsure: `bundle exec rake -T test | grep KEYWORD`
+
+### Docker requirement
+
+- Contrib/integration tests need Docker: `docker compose run --rm tracer-3.4 /bin/bash`, then run the rake task inside
+- `test:main` can run locally on any Ruby for quick feedback
```
Comment (vpellan) [inline .cursor/rules/testing.mdc:52]: In https://github.com/DataDog/dd-trace-rb/pull/5588, I added instructions to tell agents to run the tests twice: once with the earliest version we support (2.5) and once with the latest version (4.0) because it often happens that we miss regressions in older rubies.

Also, it often happens after a dependency update that cursor fails to run bundler. I think we can also add an instruction like `If Bundler fails in the container, run `bundle install` and retry once`

**Review by vpellan — APPROVED** [review]
LGTM! I had a similar PR in https://github.com/DataDog/dd-trace-rb/pull/5588 but I think yours is better, so I'll close mine

**Review by Strech — APPROVED** [review]
